rocAL/rocAL/source/pipeline/master_graph.cpp

```diff
@@ -460,7 +500,462 @@ MasterGraph::timing()
+            unsigned int fp16 = (output_data_type == RocalTensorDataType::FP16);
+
+            auto output_buffers =_ring_buffer.get_read_buffers();
+            unsigned dest_buf_offset = 0;
+            
+            if(!_output_tensor_buffer_allocated) {
+                size_t size = _output_tensor_list[0]->info().data_size() * (output_data_type == RocalTensorDataType::FP32 ? sizeof(float) : sizeof(half));
+                hipError_t status = hipMalloc(&_output_tensor_buffer, size);
+                if ((status != hipSuccess) || !_output_tensor_buffer)
+                    THROW("ROCAL::hipMalloc of size " + TOSTR(size) + " failed " + TOSTR(status))
+                _output_tensor_buffer_allocated = true;
+            }
+            
+            // copy hip buffer to out_ptr
+            // todo:: add callback routing to exchange memory pointer to avoid extra copy
+            for( auto&& out_image: output_buffers)
+            {
+                auto img_buffer = out_image;
+                auto return_status = hipMemcpy(_output_tensor_buffer, (const void *)img_buffer, sizeof(unsigned char) * n * c * h * w, hipMemcpyHostToDevice);
+                if (return_status != hipSuccess) {
+                    THROW("hipMemcpy failed with status " + TOSTR(return_status))
+                }
+                if (format == RocalTensorlayout::NHWC)
+                {
+                    HipExecCopyInt8ToNHWC(_device.resources()->hip_stream, (const void *)_output_tensor_buffer, out_ptr, dest_buf_offset, n, c, h, w,
+                                            multiplier0, multiplier1, multiplier2, offset0, offset1, offset2, reverse_channels, fp16);
+
+                }else
+                {
+                    HipExecCopyInt8ToNCHW(_device.resources()->hip_stream, (const void *)_output_tensor_buffer, out_ptr, dest_buf_offset, n, c, h, w,
+                                            multiplier0, multiplier1, multiplier2, offset0, offset1, offset2, reverse_channels, fp16);
+                }
+                dest_buf_offset += single_output_image_size;
+            }
+
+        }
+    }
+#endif
+    if((_output_tensor_list[0]->info().mem_type() == RocalMemType::HOST))
+    {
+        if(output_mem_type == RocalOutputMemType::ROCAL_MEMCPY_HOST)
+        {
+            float multiplier[3] = {multiplier0, multiplier1, multiplier2 };
+            float offset[3] = {offset0, offset1, offset2 };
+            size_t dest_buf_offset_start = 0;
+
+            auto output_buffers =_ring_buffer.get_read_buffers();
+            auto num_threads = _cpu_num_threads * 2;
+            for( auto&& out_image: output_buffers)
+            {
+                unsigned int single_image_size = w * c * h;
+                #pragma omp parallel for num_threads(num_threads)
+                for(unsigned int batchCount = 0; batchCount < n; batchCount ++)
+                {
+                    size_t dest_buf_offset = dest_buf_offset_start + single_image_size*batchCount;
```

Comment:
Please add space before and after * operator. There are few more instances of this in this file
Please check and edit

Reply:
Done